### ISSUE TITLE
Expand support for various multi-value query parameter formats

### ISSUE THREAD

**Issue by papiomytoglou** [wiremock/wiremock#2224]
### Proposal

Right now Wiremock only supports multi-value query parameters of the following format:
```
?id=1&id=2&id=3
```
That is, passing multiple values by using the same key.

However, since there is no standard dictating the format of multi-value parameters many services expect such parameters in different forms. A few examples:
```
?id[0]=1&id[1]=2&id[2]=3
?id=1,2,3
?id=1|2|3
?id=[1,2,3]
```
Support could be added so that, via a command-line argument, the user can choose either from a list of pre-existing formats or choose their own delimiters and separators. For instance:
```
# ?id[0]=1&id[1]=2&id[2]=3
java -jar wiremock-standalone.jar --multi-value-format index

# ?id=1,2,3
java -jar wiremock-standalone.jar --multi-value-format comma

# ?id=[1,2,3]
java -jar wiremock-standalone.jar --multi-value-format json

# ?id={1-2-3}
java -jar wiremock-standalone.jar --multi-value-parameter-separator "-" --multi-value-parameter-delimiters "{}"
```

Alternatively, for reasons of simplicity, some commonly used multi-value parameter formats such as the ones listed above could be supported by default in order to reduce the input needed by the user without the support for custom delimiters and separators.

### References

Documentation for the query-string TypeScript library taking a similar approach to multi-value query parameters: https://www.npmjs.com/package/query-string

Original request for support of multi-value parameters using duplicate keys: https://github.com/wiremock/wiremock/issues/398

**Comment by kapishmalik** [comment]
@tomakehurst I ll tackle this issue. You can assign this to me.

**Comment by kapishmalik** [comment]
@papiomytoglou problem is not with matcher. It is problem with way we are parsing the query parameters and its support and making available for matching.

**Comment by tomakehurst** [comment]
Since these additional formats aren't part of the http standard I suggest we add a config parameter listing all of the formats we want to enable.

**Comment by kapishmalik** [comment]
@tomakehurst  you are correct. I tried using the method jakarta.servlet.ServletRequest#getParameterMap to extract query parameters for above mentioned format. It extracts as single value instead of multiple values.
